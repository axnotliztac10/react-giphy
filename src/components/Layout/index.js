import React from 'react';
import { Flex, Box, Heading } from 'rebass';

import GlobalStyles from 'components/GlobalStyles';

export default function Layout(props) {
  const { children, heading } = props;

  return (
    <React.Fragment>
      <GlobalStyles  />
      <Flex
        alignItems="center"
        justifyContent="center"
        flexDirection="column"
      >
        <Box width="100%" p={15}>
          <Heading>{heading}</Heading>
          {children}
        </Box>
      </Flex>
    </React.Fragment>
  );
}
