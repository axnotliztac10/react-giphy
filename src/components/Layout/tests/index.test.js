import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Layout from '../index';

describe('<Layout />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Layout heading="foobar">
        <div />
      </Layout>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
