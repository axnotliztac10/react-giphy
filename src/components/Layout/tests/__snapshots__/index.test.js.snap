// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <Styled(styled.div)
    alignItems="center"
    flexDirection="column"
    justifyContent="center"
  >
    <styled.div
      p={30}
      width="100%"
    >
      <Styled(styled.div)
        justifyContent="space-between"
      >
        <Styled(Styled(styled.div))
          as="h2"
          fontSize={28}
          fontWeight="bold"
          m={0}
        >
          foobar
        </Styled(Styled(styled.div))>
        <Styled(styled.div)>
          <Link
            replace={false}
            to="/home"
          >
            Home
          </Link>
          <styled.div
            ml={10}
          >
            <Link
              replace={false}
              to="favorites"
            >
              Favorites
            </Link>
          </styled.div>
        </Styled(styled.div)>
      </Styled(styled.div)>
      <div />
    </styled.div>
  </Styled(styled.div)>
</Fragment>
`;
